src/services/GenerateVerifiableCredential.ts

```diff
@@ -227,83 +230,86 @@ export class GenerateVerifiableCredential {
-						},
-					];
-					break;
-				case "DRIVING_LICENCE": //TBC
-					credentialSubject.drivingPermit = [
-						{
-							personalNumber: documentFields.document_number,
-							expiryDate: documentFields.expiration_date,
-							issueDate: documentFields.date_of_issue,
-							issuedBy: documentFields.issuing_country, //TBC - May need to be mapped
-						},
-					];
-					break;
-				case "NATIONAL_ID": //TBC
-				credentialSubject.nationalId = [
-					{
-						personalNumber: documentFields.document_number,
-						expiryDate: documentFields.expiration_date,
-						issueDate: documentFields.date_of_issue,
-						issuedBy: documentFields.issuing_country, //TBC - May need to be mapped
-					},
-				];
-				break;
-				default:
-					throw new AppError(HttpCodesEnum.SERVER_ERROR, "Invalid documentType provided", {
-						documentType, issuingCountry,
-					});
-			}
-		}
+	  if (issuingCountry === "GBR") {
+  		switch (documentType) {
+  			case "PASSPORT":
+  				credentialSubject.passport = [
+  					{
+  						documentNumber: documentFields.document_number,
+  						expiryDate: documentFields.expiration_date,
+  						icaoIssuerCode: documentFields.issuing_country,
+  					},
+  				];
+  				break;
+  			case "DRIVING_LICENCE":
+  				credentialSubject.drivingPermit = [
+  					{
+  						personalNumber: documentFields.document_number,
+  						expiryDate: documentFields.expiration_date,
+  						issueDate: documentFields.date_of_issue,
+  						issuedBy: documentFields.issuing_authority,
+  						fullAddress: documentFields.formatted_address,
+  					},
+  				];
+  				break;
+  			case "RESIDENCE_PERMIT":
+  				credentialSubject.residencePermit = [
+  					{
+  						documentNumber: documentFields.document_number,
+  						expiryDate: documentFields.expiration_date,
+  						issueDate: documentFields.date_of_issue,
+  						icaoIssuerCode: documentFields.issuing_country,
+  					},
+  				];
+  				break;
+  			default:
+  				throw new AppError(HttpCodesEnum.SERVER_ERROR, "Invalid documentType provided", {
+  					documentType, issuingCountry,
+  				});
+  		}
+  	} else {
+  		switch (documentType) {
+  			case "PASSPORT":
+  				credentialSubject.passport = [
+  					{
+  						documentNumber: documentFields.document_number,
+  						expiryDate: documentFields.expiration_date,
+  						icaoIssuerCode: documentFields.issuing_country,
+  					},
+  				];
+  				break;
+			case "DRIVING_LICENCE":
+				const countryDetails = EU_DL_COUNTRIES.find(country => country.alpha3code === documentFields.issuing_country);
+  				credentialSubject.drivingPermit = [
+  					{
+  						personalNumber: documentFields.document_number,
+  						expiryDate: documentFields.expiration_date,
+  						issueDate: documentFields.date_of_issue,
+  						issuedBy: documentFields.place_of_issue,
+  						//issuingCountry: documentFields.issuing_country,
+						issuingCountry: countryDetails ? countryDetails.alpha2code : "n/a"
```

Comment:
are we okay to return "n/a" for a country not present in the map in the VC being returned?

Reply:
Checking with Anand on the expected result for this case.

Reply:
throwing an error if the mapping is not found for the country.